src/CMakeLists.txt

```diff
@@ -482,7 +482,17 @@ if( MIOPEN_BACKEND STREQUAL "HIPOC" OR MIOPEN_BACKEND STREQUAL "HIP")
         )
 endif()
 
-if( MIOPEN_BACKEND MATCHES "OpenCL" OR MIOPEN_BACKEND STREQUAL "HIPOC" OR MIOPEN_BACKEND STREQUAL "HIP")
+if( MIOPEN_BACKEND STREQUAL "NOGPU")
+    list(APPEND MIOpen_Source
+        hip/hiperrors.cpp
+        nogpu/handle.cpp
```

Comment:
[Notice] AFAICS in this PR does not use a mechanism designed to add new backends -- I mean the HandleImpl machinery, -- in the originally intended way. So I am curious, why.

However, I do not have time to consider this issue in detail. If it doesn't break anything, and it works for you, then okay. For now.

Reply:
The PR does introduce a new implementation [here](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/38017007a19e34f005458ddeb505f703b65c78c7/src/include/miopen/nogpu/handle_impl.hpp#L1) which has been moved out of the cpp file so that the client program can include the header file and override the behavior when desired.